src/communication/mpi.cpp

```diff
@@ -8,8 +8,8 @@ namespace mpi {
 
 // global state
 namespace state {
```

Comment:
What do the initial values here mean?

Reply:
They are sane defaults (1 mpi rank with rank 0), to make the unit tests pass when compiled with MPI.  The unit tests don't initialize MPI, so the default signed values of -1 were being converted to there unsigned version in the domain_decomposition unit test (the one in `tests.exe`, not the `global_communication.exe` test runner which initializes MPI).

This is not a permanent solution: @halfflat and I plan to use "executors", which will be more flexible than the current hard-coded compile time approach.

Reply:
Ah -- I see, if the init is never called, it pretends to be rank 0 of a world size 1.
Could we just comment that for the time being, so that if someone trips over it, they won't be confused? -1, -1 says to me that if init is never called, mark it with illegal values -- but the new version has unclear, far away implications.

Reply:
Ah -- I see, if the init is never called, it pretends to be rank 0 of a world size 1.
Could we just comment that for the time being, so that if someone trips over it, they won't be confused? -1, -1 says to me that if init is never called, mark it with illegal values -- but the new version has unclear, far away implications.

Reply:
I have added a comment, marking it as a TODO that should be removed when we add the run time executors.